jbpm-flow/src/main/java/org/jbpm/process/core/timer/impl/ThreadPoolSchedulerService.java

```diff
@@ -93,18 +92,11 @@ public JobHandle scheduleJob(Job job, JobContext ctx, Trigger trigger) {
 
         Date date = trigger.hasNextFireTime();
         if ( date != null ) {
-            String jobname = null;
-            if (ctx instanceof ProcessJobContext) {
-                ProcessJobContext processCtx = (ProcessJobContext) ctx;
-                jobname = processCtx.getSessionId() + "-" + processCtx.getProcessInstanceId() + "-" + processCtx.getTimer().getId();
-                if (processCtx instanceof StartProcessJobContext) {
-                    jobname = "StartProcess-"+((StartProcessJobContext) processCtx).getProcessId()+ "-" + processCtx.getTimer().getId();
-                }
-                if (activeTimer.containsKey(jobname)) {
-                    return activeTimer.get(jobname);
-                }
-            
+            String jobname = QuartzNameHelper.getJobName(ctx);
+            if (activeTimer.containsKey(jobname)) {
+                return activeTimer.get(jobname);
```

Comment:
This replacement is not exactly as it was before, as only has to return the activeTime if ctx is instance of ProcessJobContext (missing this condition for all the block, though it's evaluated also inside helper class), wdyt?

Reply:
I think regardless of the job type, it should not be created again if already exist a unique job name for it